tests/components/camera/test_init.py

```diff
@@ -294,29 +294,73 @@ async def test_websocket_get_prefs(hass, hass_ws_client, mock_camera):
     assert msg["success"]
 
 
-async def test_websocket_update_prefs(
-    hass, hass_ws_client, mock_camera, setup_camera_prefs
-):
-    """Test updating preference."""
-    await async_setup_component(hass, "camera", {})
-    assert setup_camera_prefs[PREF_PRELOAD_STREAM]
+async def test_websocket_update_prefs(hass, hass_ws_client, mock_camera):
+    """Test updating camera preferences."""
+
+    legacy_camera_prefs_by_entity = hass.data[DATA_CAMERA_PREFS]._prefs
+
+    assert "demo.uniquecamera" not in legacy_camera_prefs_by_entity
+
     client = await hass_ws_client(hass)
     await client.send_json(
         {
             "id": 8,
             "type": "camera/update_prefs",
-            "entity_id": "camera.demo_camera",
-            "preload_stream": False,
+            "entity_id": "camera.demo_uniquecamera",
+            "preload_stream": True,
         }
     )
     response = await client.receive_json()
-
     assert response["success"]
-    assert not setup_camera_prefs[PREF_PRELOAD_STREAM]
-    assert (
-        response["result"][PREF_PRELOAD_STREAM]
-        == setup_camera_prefs[PREF_PRELOAD_STREAM]
+
+    # preload_stream entry for this camera should have been added
+    assert "camera.demo_uniquecamera" in legacy_camera_prefs_by_entity
+    legacy_camera_prefs = legacy_camera_prefs_by_entity["camera.demo_uniquecamera"]
+    assert legacy_camera_prefs[PREF_PRELOAD_STREAM] is True
+    assert response["result"][PREF_PRELOAD_STREAM] is True
+
+    # Try sending orientation update for entity not in entity registry
```

Comment:
I suggest splitting this up into a few different tests (e.g. with unique id, without unique id, etc) -- just for making it easier to follow exactly what state is under test for each scenario.

Reply:
Ended up splitting it into two tests, one for preload and one for orientation (the case where both values are updated simultaneously is no longer exercised, but the frontend does not send that anyway).